### ISSUE TITLE
False positive `SwallowedException` when exception is used as a receiver

### ISSUE THREAD

**Issue by matejdro** [detekt/detekt#7281]
## Expected Behavior
`SwallowedException` should not trigger

## Observed Behavior
`SwallowedException` triggers

## Steps to Reproduce

Analyze following code:

```
private fun Exception.transformException(): Exception {
   return this
}

private fun demo() {
   try {
      TODO()
   } catch (e: Exception) {
      throw e.transformException()
   }
}
```

## Context

`Exception.transformException()` is functionally equivalent to the `transformException(source: Exception)` (which does not trigger this rule), so the rule should be satisfied with it.

## Your Environment

* Version of detekt used: 1.23.6
